HP-UX headers before HP-UX 11.31 (11iv3) lack definition of SEM_FAILED.

--- Modules/_multiprocessing/semaphore.c.orig	2009-09-25 15:21:22 +0200
+++ Modules/_multiprocessing/semaphore.c	2009-09-25 15:21:58 +0200
@@ -190,6 +190,10 @@
  * Unix definitions
  */
 
+#ifndef SEM_FAILED
+#define SEM_FAILED (SEM_HANDLE)-1
+#endif
+
 #define SEM_CLEAR_ERROR()
 #define SEM_GET_LAST_ERROR() 0
 #define SEM_CREATE(name, val, max) sem_open(name, O_CREAT | O_EXCL, 0600, val)
